include/zephyr/smf.h

```diff
@@ -47,33 +31,14 @@
  * @param _parent  State parent object or NULL
  * @param _initial State initial transition object or NULL
  */
-#define SMF_CREATE_STATE(_entry, _run, _exit, _parent, _initial) \
-{ \
-	.entry   = _entry,  \
-	.run     = _run,    \
-	.exit    = _exit,   \
-	.parent  = _parent, \
-	.initial = _initial \
+#define SMF_CREATE_STATE(_entry, _run, _exit, _parent, _initial)               \
+{                                                                              \
+	.entry = _entry,                                                       \
+	.run   = _run,                                                         \
+	.exit  = _exit,                                                        \
+	COND_CODE_1(CONFIG_SMF_ANCESTOR_SUPPORT, (.parent = _parent,), ())     \
```

Comment:
could be further simplified
```suggestion
	IF_ENABLED(CONFIG_SMF_ANCESTOR_SUPPORT, (.parent = _parent,))     \
```